### PR TITLE
Add workflow to manage repository labels

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This pull request adds a new GitHub Actions workflow and configuration file to manage repository labels.

## 💭 Motivation and context ##

This adds a new workflow using the [crazy-max/ghaction-github-labeler](https://github.com/crazy-max/ghaction-github-labeler) Action to manage a repository's labels. This will ensure that any repository derived from a skeleton has a consistent starting set of labels. Per-repository labels can then be layered in using this same configuration to ensure they are consistent. The labels in the configuration file are based on the configured labels in [cisagov/.github](https://github.com/cisagov/.github).

This workflow is currently only run on changes to the configuration file or the workflow. We could also configure it to run on changes to labels per the [documentation](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#label).

## 🧪 Testing ##

Automated tests pass.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

## ✅ Post-merge checklist ##

- [x] ~~Archive~~ Delete [cisagov/github-label-management](https://github.com/cisagov/github-label-management)
- [x] Close https://github.com/cisagov/.github/pull/29

### CODE REVIEW

**Review by jsf9k — APPROVED** [review]
Great!  Once this is merged we can archive [cisagov/github-label-management](https://github.com/cisagov/github-label-management), correct?

**Comment by mcdonnnj** [comment]
> Great! Once this is merged we can archive [cisagov/github-label-management](https://github.com/cisagov/github-label-management), correct?

Yep. I will also close https://github.com/cisagov/.github/pull/29 as it is no longer needed.

Edit:
Added those two tasks as post-merge items

File: .github/workflows/sync-labels.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Sync repository labels
+        if: success()
+        uses: crazy-max/ghaction-github-labeler@v4
+        with:
+          # This is a hideous ternary equivalent so we only do a dry run unless
+          # this workflow is triggered by the develop branch.
+          dry-run: ${{ github.ref_name == 'develop' && 'false' || 'true' }}
```
Comment (dav3r) [inline .github/workflows/sync-labels.yml:29]: I've seen worse in our Terraform code - you're not even in the Top 10 Most Hideous with this. 😱

**Review by dav3r — APPROVED** [review]
Strong work! 👍

**Comment by jsf9k** [comment]
Requesting re-review from @dav3r since I removed a duplicate label and added a new one.  @mcdonnnj, can you also take a look?

**Review by dav3r — APPROVED** [review]
Testing- is this thing on? 🎙️

**Pull request merged**